node/pkg/processor/processor.go

```diff
@@ -241,33 +243,34 @@ func (p *Processor) Run(ctx context.Context) error {
 	}
 
 	// Start the routine to do housekeeping tasks that don't need to be distributed to the workers.
-	var w sync.WaitGroup
-	w.Add(1)
-	go func(ctx context.Context) {
-		err := p.runHousekeeper(ctx)
-		if err != nil {
-			p.logger.Error("housekeeper failed", zap.Error(err))
-		}
-	}(ctx)
+	common.RunWithScissors(ctx, errC, "processor_housekeeper",
+		func(ctx context.Context) error {
+			err := p.runHousekeeper(ctx)
+			if err != nil {
+				errC <- fmt.Errorf("processor housekeeper failed: %v", err)
+			}
+			return err
```

Comment:
I think there's no need to wrap this in an extra function. The way it is right now will actually result in two write attempts to errC: One by this function and another one by RunWithScissors. Each channel should only have one writer and one consumer as a general practice. 

Instead `common.RunWithScissors(ctx, errC, "processor_housekeeper", p.runHousekeeper)` should work, no? 

PS: Still not sure if panics should be caught here or just propagated (see my other comment).